esp-hal-common/src/gpio.rs

```diff
@@ -976,6 +1000,19 @@ macro_rules! impl_output {
                 self
             }
 
+            fn disconnect_peripheral_from_output(
+                &mut self,
+            ) -> &mut Self {
+                // Reset this GPIO to plain GPIO
+                self.set_alternate_function(AlternateFunction::Function2);
```

Comment:
`Function2` is the GPIO function ONLY on ESP32 - on other chips it's currently Function1 but for future chips it might be even some other function. That's what the macro parameter `$gpio_function` is for